src/System.Reflection.TypeExtensions/tests/TypeTests.cs

```diff
@@ -2,24 +2,905 @@
+            yield return new object[] { BindingFlags.DeclaredOnly | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, new string[0] };
+            yield return new object[] { BindingFlags.IgnoreCase | BindingFlags.DeclaredOnly | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, new string[0] };
+
+            yield return new object[] { BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, expectedAll };
+            yield return new object[] { BindingFlags.IgnoreCase | BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, expectedAll };
+            yield return new object[] { BindingFlags.DeclaredOnly | BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, expectedAll };
+            yield return new object[] { BindingFlags.IgnoreCase | BindingFlags.DeclaredOnly | BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, expectedAll };
+
+            yield return new object[] { BindingFlags.FlattenHierarchy, new string[0] };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetEvents_TestData))]
+        public void GetEvents(BindingFlags bindingAttributes, string[] expectedNames)
+        {
+            if (bindingAttributes == BindingFlags.Default)
+            {
+                string[] eventNames = typeof(Cat<int>).GetEvents().Select(eventInfo => eventInfo.Name).ToArray();
+                Assert.All(expectedNames, name => eventNames.Contains(name));
+            }
+            else
+            {
+                string[] eventNames = typeof(Cat<int>).GetEvents(bindingAttributes).Select(eventInfo => eventInfo.Name).ToArray();
+                Assert.All(expectedNames, name => eventNames.Contains(name));
+            }
+        }
+
+        public static IEnumerable<object[]> GetFields_TestData()
+        {
+            yield return new object[] { typeof(GenericClassUsingNestedInterfaces<string, int>), new string[] { "FieldZero", "FieldOne", "FieldTwo", "FieldThree" } };
+            yield return new object[] { typeof(GenericStructWithInterface<string>), new string[] { "field", "field2" } };
+            yield return new object[] { typeof(NonGenericClassWithGenericInterface), new string[] { "field" } };
+            yield return new object[] { typeof(GenericStruct2TP<int, string>), new string[] { "field", "field2" } };
+            yield return new object[] { typeof(GenericClassWithVarArgMethod<>), new string[] { "field" } };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetFields_TestData))]
+        public void GetFields(Type type, string[] expectedNames)
+        {
+            string[] fieldNames = type.GetFields().Select(field => field.Name).ToArray();
+            Assert.All(expectedNames, name => fieldNames.Contains(name));
```

Comment:
Its because BindingFlags.Default does not actually equal the default flags used by the methods, I've replaced this with `BindingFlags.Public |  BindingFlags.Instance | BindingFlags.Static`